force-app/classes/src/Assertions.cls

```diff
@@ -6,76 +6,137 @@
  */
 @IsTest
 public class Assertions {
-  public virtual class SystemAsserter {
-    public virtual void assertEquals(
-      Object expected,
-      Object actual,
-      String message
-    ) {
-      System.assertEquals(expected, actual, message);
+  public interface MethodSpyAssertions {
+    /**
+     * Assert that a method spy has never been called
+     */
+    void hasNotBeenCalled();
+
+    /**
+     * Assert that a method spy has been called at least one time with any parameters
+     */
+    void hasBeenCalled();
+
+    /**
+     * Assert that a method spy has been called exactly N times with any parameters
+     *
+     * @param count number of times the method spy should have been called
+     */
+    void hasBeenCalledTimes(final Integer count);
+
+    /**
+     * Assert that a method spy has been called at least one time with matching parameters
+     *
+     * @param params params that should match one call paramters
+     * @see          Params
+     */
+    void hasBeenCalledWith(final Params params);
+
+    /**
+     * Assert that a method spy has been lastly called with matching parameters
+     *
+     * @param params params that should match last call paramters
+     * @see          Params
+     */
+    void hasBeenLastCalledWith(final Params params);
+  }
+
+  public interface Asserter {
+    void isTrue(Boolean value, String message);
+    void isFalse(Boolean value, String message);
+  }
+
+  /**
+   * Starts to build assertions on a method spy
+   *
+   * @param spy the MethodSpy to assert
+   * @see       MethodSpy
+   */
+  public static MethodSpyAssertions assertThat(MethodSpy spy) {
+    return assertThat(spy, METHOD_SPY_ASSERTER);
+  }
+
+  /**
+   * Starts to build assertions on a method spy using a custom asserter
+   *
+   * @param spy      the MethodSpy to assert
+   * @param asserter the Assert to use during assertions
+   */
+  public static MethodSpyAssertions assertThat(
+    MethodSpy spy,
+    Asserter asserter
+  ) {
+    return new DefaultMethodSpyAssertions(spy, asserter);
+  }
+
+  @TestVisible
+  private class MethodSpyAsserter implements Asserter {
+    public void isTrue(Boolean value, String message) {
+      if (!value) {
+        throw new AssertException(message);
+      }
```

Comment:
**suggestion:** alternative solution using new [Assert class](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_class_System_Assert.htm#apex_System_Assert_isTrue)

```suggestion
      Assert.isTrue(value, message);
```

Reply:
I will do that, though I have a comment
I tried this and it's localising error message in a weird way like this

```
System.AssertException: Assertion Failed: Échec de l'assertion: Method insertSObject was not called with [(Opportunity:{AccountId=001000000000000AAB})]
method call history:
   #2 insertSObject([(Opportunity:{AccountId=001000000000000AAA})])
   #1 insertSObject([(Account:{NumberOfEmployees=10})])
```

Reply:
I think it is because we do not have added the default language in our `project-scratch-def.json`
we should add `"language": "en_US",` between `feature` and `settings` attributes in that file
It makes me think we should add a card to localize our error message (later)